sympy/core/numbers.py

```diff
@@ -2469,6 +2471,13 @@ def _eval_power(self, expt):
             return S.NaN
         if expt is S.ComplexInfinity:
             return S.NaN
+        if expt.is_real is False:
+            if re(expt).is_positive:
```

Comment:
``re`` is a function that is not quite trivial. Perhaps it would be better to call it just once.